src/Guide/Types/Core.hs

```diff
@@ -131,6 +131,12 @@ instance A.ToJSON ItemKind where
   toJSON Other = A.object [
     "tag"      A..= ("Other" :: Text) ]
 
+instance A.FromJSON ItemKind where
+  parseJSON = A.withObject "ItemKind" $ \o
```

Comment:
This doesn't work.

`<|>` picks the first parser that succeeds. If there's a field called `"contents"`, the first parser will always succeed, and the second parser will never be chosen.

You want this:

```haskell
  parseJSON = A.withObject "ItemKind" $ \o ->
    o A..: "tag" >>= \case
      "Library" -> Library <$> o A..: "contents"
      "Tool"    -> Tool <$> o A..: "contents"
      "Other"   -> pure Other
      tag       -> fail ("unknown tag " ++ show tag)
```